app/routes/task_routes.py

```diff
@@ -1 +1,129 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response, Response, json
+from app import db
+from app.models.task import Task
+from datetime import datetime
+import os
+import requests
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks") 
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body:
+        return{"details": "Invalid data"}, 400
+ 
+    new_task = Task(
+        title=request_body["title"],
+        description=request_body["description"],
+        completed_at=None,  
+        is_complete=False   
+    )
+    
+    db.session.add(new_task)
+    db.session.commit()
+    
+    return {"task": new_task.to_dict()}, 201
+
+@tasks_bp.get("")
+def get_tasks():
+    tasks = Task.query.all()
+    sort_param = request.args.get("sort")
+    
+    sort_options ={
+        "asc": Task.title.asc(),
+        "desc": Task.title.desc()
+    }
+    
+    query = db.select(Task)
+    if sort_param in sort_options:
+        query= query.order_by(sort_options[sort_param])
+        
+    tasks = db.session.scalars(query).all()
```

Comment:
This is the syntax we should be using for querying any time we are trying to query for a result that could be more than a single record and may be applying filters and sorting to the results. 

When we know we only want a single record and have the primary key, there is a valid shorthand syntax through the `db.session` object:
```py
db.session.get(Task, task_id)
```